File: app/src/main/java/com/pawsitive/app/auth/RoleSelectionActivity.java
package com.pawsitive.app.auth;

import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.widget.Button;
import android.widget.EditText;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.pawsitive.app.R;
import com.pawsitive.app.ngo.NGORegistrationActivity;
import com.pawsitive.app.user.UserProfileActivity;

public class RoleSelectionActivity extends AppCompatActivity {

    private EditText etFullName;
    private RadioGroup rgRoles;
    private RadioButton rbUser, rbNgo;
    private Button btnContinue;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_role_selection);

        etFullName = findViewById(R.id.etFullName);
        rgRoles = findViewById(R.id.rgRoles);
        rbUser = findViewById(R.id.rbUser);
        rbNgo = findViewById(R.id.rbNgo);
        btnContinue = findViewById(R.id.btnContinueRole);

        // Get credentials from SignupActivity
        String email = getIntent().getStringExtra("email");
        String password = getIntent().getStringExtra("password");

        btnContinue.setOnClickListener(v -> {
            String fullName = etFullName.getText().toString().trim();
            int selectedId = rgRoles.getCheckedRadioButtonId();

            if (TextUtils.isEmpty(fullName)) {
                etFullName.setError("Full name is required");
                etFullName.requestFocus();
                return;
            }

            if (selectedId == -1) {
                Toast.makeText(RoleSelectionActivity.this, "Please select a role", Toast.LENGTH_SHORT).show();
                return;
            }

            Intent intent;
            if (selectedId == R.id.rbUser) {
                intent = new Intent(RoleSelectionActivity.this, UserProfileActivity.class);
            } else if (selectedId == R.id.rbNgo) {
                intent = new Intent(RoleSelectionActivity.this, NGORegistrationActivity.class);
            } else {
                return;
            }
            
            intent.putExtra("full_name", fullName);
            intent.putExtra("email", email);
            intent.putExtra("password", password);
            startActivity(intent);
            finish();
        });
    }
}

File: app/src/main/java/com/example/pawsitive/auth/RoleSelectionActivity.kt
package com.example.pawsitive.auth

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.RadioButton
import android.widget.RadioGroup
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.pawsitive.R
import com.example.pawsitive.ngo.NGORegistrationActivity
import com.example.pawsitive.user.UserProfileActivity

class RoleSelectionActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_role_selection)

        val fullName = intent.getStringExtra("full_name") ?: ""
        val email = intent.getStringExtra("email") ?: ""

        val rgRoles: RadioGroup = findViewById(R.id.rgRoles)
        val rbUser: RadioButton = findViewById(R.id.rbUser)
        val rbNgo: RadioButton = findViewById(R.id.rbNgo)
        val btnContinue: Button = findViewById(R.id.btnContinueRole)

        btnContinue.setOnClickListener {
            val selectedId = rgRoles.checkedRadioButtonId
            if (selectedId == -1) {
                Toast.makeText(this, "Please select a role", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            if (selectedId == rbUser.id) {
                val intent = Intent(this, UserProfileActivity::class.java).apply {
                    putExtra("full_name", fullName)
                    putExtra("email", email)
                }
                startActivity(intent)
            } else if (selectedId == rbNgo.id) {
                val intent = Intent(this, NGORegistrationActivity::class.java).apply {
                    putExtra("email", email)
                }
                startActivity(intent)
            }

            finish()
        }
    }
}
